lib/messages.js

```diff
@@ -101,6 +102,11 @@ const connection = {
       id.encode(state, m.handshakeId)
     }
 
+    if (m.publicKey) {
+      flags |= 2
+      publicKey.encode(state, m.publicKey)
```

Comment:
Now it should be backward compat. Maybe preencode and encode should not have the conditionals but still change the flag